src/battle_script_commands.c

```diff
@@ -5537,6 +5538,16 @@ static void Cmd_moveend(void)
             }
             gBattleScripting.moveendState++;
             break;
+        case MOVEEND_NUM_HITS:
+            if (gBattlerAttacker != gBattlerTarget
+                && GetBattlerSide(gBattlerAttacker) != GetBattlerSide(gBattlerTarget)
```

Comment:
Remove this and the line above. Confusion damage counts. The three checks after this are enough (although I would move the `gBattleMoves[gCurrentMove].split != SPLIT_STATUS` to it's own line)

Reply:
> Confusion damage counts
Does it? I know Bulbagarden says it does but both Showdown and research from DaWoblefet say it doesn't.

Reply:
All done. 

For Substitute I already had a test specifically to test the edge case.